### PR TITLE
Bump and rename siduck76-st > st-snazzy

### PR DESCRIPTION
**Pull request by pouya-abbassi**
The username has changed, and the name/description of the project.

Also the nixpkgs version was too old.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I have made a flake, used the nixpkgs on my machine as input, and ran `nix build` to get the result.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/st/st-snazzy/package.nix [hunk 1]
```diff
@@ -42,10 +44,10 @@ stdenv.mkDerivation rec {
...
 
   meta = with lib; {
-    homepage = "https://github.com/siduck76/st";
-    description = "Fork of st with many add-ons";
+    homepage = "https://github.com/siduck/st";
+    description = "snazzy terminal (suckless + beautiful)";
     license = licenses.mit;
-    maintainers = with maintainers; [ ];
+    maintainers = with maintainers; [ AndersonTorres ];
     platforms = platforms.linux;
```
Comment (LordGrimmauld) [inline pkgs/by-name/st/st-snazzy/package.nix:51]: You are refactoring the whole package. Drop the `with lib;`, its deprecated for a while now.

**Review by LordGrimmauld — CHANGES_REQUESTED** [review]
You'll have to squash your commits and put messages according to CONTRIBUTING.md/Commit conventions

File: pkgs/by-name/st/st-snazzy/package.nix (hunk 1 above)
Comment (LordGrimmauld) [reply pkgs/by-name/st/st-snazzy/package.nix:51]: Sidenote: Don't readd AndersonTorresas maintainer without asking! This is a no-go.

Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix]: I don't know why I thought they were the original maintainer.
Thanks for review.

**Review by LordGrimmauld — CHANGES_REQUESTED** [review]
Please do the maintainer addition to maintainer-list *before* the package change so our CI doesn't get confused.

File: pkgs/by-name/st/st-snazzy/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    glib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    TERMINFO=$out/share/terminfo make install PREFIX=$out
+
+    runHook postInstall
+  '';
```
[the remark this answers is not in the document]
Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix:45]: It doesn't compile when I apply this change. I get the following error:
```
       > mkdir -p /usr/local/bin
       > mkdir: cannot create directory '/usr': Permission denied
       > make: *** [Makefile:46: install] Error 1
```
I'm relatively new to Nix, so I don't quite understand why you made that suggestion and what the difference is between the two forms.
I have also tried this and got the same result:
```
  env.TERMINFO = "${placeholder "out"}/share/terminfo make install PREFIX=${placeholder "out"}";
```

[the remark this answers is not in the document]
Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix:45]: I tried `env.PREFIX = "/tmp";` and didn't work.
Your comment worked. Thanks!

File: pkgs/by-name/st/st-snazzy/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  src = fetchFromGitHub {
+    owner = "siduck";
+    repo = "st";
+    rev = "a7582f96afdee6bf0793cd0d8f84b755fd6aabf6";
+    hash = "sha256-wohkmDsm26kqFGQKuY6NuBQsifT7nZNgrLqLFsU+Vog=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
```
[the remark this answers is not in the document]
Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix]: `nixfmt 'pkgs/by-name/st/st-snazzy/package.nix'` doesn't like this.

**Review by ghost — APPROVED** [review]

File: pkgs/by-name/st/st-snazzy/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  libXft,
+  ncurses,
+  gd,
+  glib,
+  pkg-config,
+}:
+
+stdenv.mkDerivation {
+  pname = "st-snazzy";
+  version = "0.8.5";
```
Comment (SuperSandro2000) [inline pkgs/by-name/st/st-snazzy/package.nix:17]: How does that version correlate with the commit below?

Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix:17]: I took it from the default.nix in the original repo.
https://github.com/siduck/st/blob/main/default.nix
There's no other mention of version anywhere else in the project.
Do you suggest something like `version = "2024-09-06";` (date of the last commit)?

Comment (pouya-abbassi) [reply pkgs/by-name/st/st-snazzy/package.nix]: asked on Matrix and people suggested `version = "0.8.5-unstable-2024-09-06";`, so I made the change.

**Pull request merged**